python_modules/dagster/dagster/core/storage/fs_io_manager.py

```diff
@@ -119,7 +120,20 @@ def handle_output(self, context, obj):
         mkdir_p(os.path.dirname(filepath))
 
         with open(filepath, self.write_mode) as write_obj:
-            pickle.dump(obj, write_obj, PICKLE_PROTOCOL)
+            try:
+                pickle.dump(obj, write_obj, PICKLE_PROTOCOL)
+            except (AttributeError, RecursionError, ImportError, pickle.PicklingError):
+                executor = context.step_context.pipeline_def.mode_definitions[0].executor_defs[0]
+
+                raise DagsterInvariantViolationError(
+                    f"Object {obj} is not picklable. Use the mem_io_manager with an in process "
+                    "executor to avoid pickling outputs. You are currently using the "
+                    f"fs_io_manager and the {executor.name}. \n"
+                    "For more information on io managers, visit "
+                    "https://docs.dagster.io/concepts/io-management/io-managers \n"
+                    "For more information on executors, vist "
+                    "https://docs.dagster.io/deployment/executors#overview"
```

Comment:
This is an awesome error message

Reply:
I shifted the language a bit to suggest the in memory and in process as a potential option. lmk how it reads